### PR TITLE
adds devfile provider for dev catalog

### PR DESCRIPTION
**Pull request by invincibleJai**
**Fixes**: 
https://issues.redhat.com/browse/ODC-5729

**Analysis / Root cause**: 

**Solution Description**: 
adds devfile provider for dev catalog so that user can see type `devfiles` in the catalog and create one

**Screen shots / Gifs for design review**: 
@openshift/team-devconsole-ux @lwrigh 

image

image

image

image

**Review commit**: https://github.com/openshift/console/commit/9c1b913b9d2bc9ab74f3d02497dc8ff29cab0f9a

**Depends on**:  https://github.com/openshift/console/pull/8794

**Browser conformance**: 

- [x] Chrome
- [ ] Firefox
- [ ] Safari
- [ ] Edge

### CODE REVIEW

**Comment by invincibleJai** [comment]
/hold 

adding hold as depends on https://github.com/openshift/console/pull/8794

**Comment by lwrigh** [comment]
This looks good to me!

File: frontend/packages/dev-console/src/components/catalog/providers/useDevfile.tsx [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+  const [loadedError, setLoadedError] = React.useState<APIError>();
+  const { t } = useTranslation();
+
+  React.useEffect(() => {
+    const payload = {
+      registry: 'pkg/devfile/sample-placeholder.json',
+    };
+    coFetchJSON
+      .put('/api/devfile/samples', payload)
+      .then(setDevfileSamples)
```
Comment (rohitkrai03) [inline frontend/packages/dev-console/src/components/catalog/providers/useDevfile.tsx:68]: Should check if component is still mounted before setting state.

**Review by rohitkrai03 — CHANGES_REQUESTED** [review]

**Comment by beaumorley** [comment]
Hi @invincibleJai. One question. On the Developer Catalog there is a (?) next to Type. This is a popover with a definition of all the types. Do you know if this issue includes adding Devfiles to this? I did not see it in the Epic.

**Comment by invincibleJai** [comment]
> Hi @invincibleJai. One question. On the Developer Catalog there is a (?) next to Type. This is a popover with a definition of all the types. Do you know if this issue includes adding Devfiles to this? I did not see it in the Epic.

yes @beaumorley, have handled it as part of the same PR. If you think there is a need to change the text for devFile LMK or can log a ticket to update it.

image

**Comment by christianvogt** [comment]
We had a discussion on this scenario in the devfile cabal. Please connect with @rohitkrai03. As the scenario for changing the git url and using the sample as a stack wasn't considered by the devfile team.

**Pull request merged**